### PR TITLE
Doc start

### PR DESCRIPTION
**Pull request by ianrice07**
Initial outline with some actual documentation filled in.  It'll be used as a template for the rest of the docs.

### CODE REVIEW

File: packages/browser/README.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+```js
+// Import the attachment types from @backtrace/browser with your favoriate package manager.
+import { BacktraceUint8ArrayAttachment, BacktraceStringAttachment  } from "@backtrace/browser";
+
+// BacktraceStringAttachment should be used for text object like a log file for instance
+var attachment1 = new BacktraceStringAttachment("logfile.txt", "This is the start of my log")
+
+// BacktraceUint8ArrayAttachment should be used for binary files
+var attachment2 = new BacktraceUint8ArrayAttachment("output.dmp", new Uint8Array(2));
```
Comment (konraddysput) [inline packages/browser/README.md:49]: actually I'm not sure if this is the best example. You won't send minidump from web. Maybe instead we can send connection buffer or something like that?

File: packages/browser/README.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+// Initialize the client
+const client = BacktraceClient.builder(options).build();
+
+// When an error occurs
+client.send(new Error("Something broke!"));
+
+```
+## Features
+
+### File Attachments
```
Comment (konraddysput) [inline packages/browser/README.md:37]: Shouldn't we start with attributes?

File: packages/browser/README.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+|-|-|-|-|
+|`metricsSubmissionUrl`|String|Metrics server hostname|`https://events.backtrace.io`|
+|`enable`|Boolean|Determines if the metrics support is enabled|`true`|
+|`autoSendInterval`|Integer|Indicates how often crash free metrics are sent to Backtrace. The interval is a value in ms. The default is 30 minutes|`1800000`|
+|`size`|Integer|Indicates how many events the metrics storage can store before auto submission|`50`|
+
+### Breadcrumbs
+Feature in developement
+
+### Callback Methods
```
Comment (rick-bt) [inline packages/browser/README.md:95]: I would tentatively assert:
Callback methods would be part of Documentation.
"Features" would be info relating to how to support a Backtrace system/UI feature.
"Documentation" would be info on just the SDK - anything mechanical that happens prior to the report being sent.

File: packages/browser/README.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    return data;
+}
+
+const test: BacktraceConfiguration = {
+    name: "blah",
+    version: "1.2.3",
+    url: "https://submit.backtrace.io/<universe>/<token>/json",
+    beforeSend: myBeforeSend
+}
+```
```
Comment (rick-bt) [inline packages/browser/README.md:117]: I would tentatively assert:
Callback methods would be part of Documentation.
"Features" would be info relating to how to support a Backtrace system/UI feature.
"Documentation" would be info on just the SDK - anything mechanical that happens prior to the report being sent.

File: packages/browser/README.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+import { BacktraceClient } from "@backtrace/browser";
+
+// BacktraceClientOptions
+const options = {
+    // Name of the website/application
+    name: "MyWebPage",
+    // Version of the website
+    version: "1.2.3",  
+    // Submission url
+    url: "https://submit.backtrace.io/<universe>/<token>/json",
```
Comment (rick-bt) [inline packages/browser/README.md:25]: Where can we get universe, token?
```
// Universe is the subdomain of Backtrace hosted instance (universe.backtrace.io)
// Token can be found in Project Settings/Submission tokens
```

File: packages/browser/README.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+1. [Features](#features)
+    - [File Attachments](#file-attachments)
+    - [Metrics](#metrics)
+    - [Breadcrumbs](#breadcrumbs)
+    - [Callback Methods](#callback-methods)
+1. [Documentation](#documentation)
+
+## Integration Example
+```js
+// Import the BacktraceClient from @backtrace/browser with your favoriate package manager.
```
Comment (rick-bt) [inline packages/browser/README.md:15]: <sp>

File: packages/browser/README.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+Backtrace error reporting tool for browser side JavaScript
+ 
+## Table of Contents
+1. [Integration Example](#integration-example)
+1. [Features](#features)
+    - [File Attachments](#file-attachments)
+    - [Metrics](#metrics)
+    - [Breadcrumbs](#breadcrumbs)
+    - [Callback Methods](#callback-methods)
+1. [Documentation](#documentation)
```
Comment (rick-bt) [inline packages/browser/README.md:11]: Thinking about an alt layout that divides between SDK implementation and Backtrace feature implementation...
* SDK Documentation
  * Initial integration - Getting your first errors
  * BacktraceClient configuration
  * Error report modification _(callback methods here)_
* Error Reporting features
  * Attributes
  * Breadcrumbs
  * File attachments
  * Application stability metrics

File: packages/browser/README.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
```
Comment (rick-bt) [inline packages/browser/README.md:12]: We also need to prompt the user to integrate source maps and source code...

File: packages/browser/README.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+// BacktraceClientOptions
+const options = {
+    // Name of the website/application
+    name: "MyWebPage",
+    // Version of the website
+    version: "1.2.3",  
+    // Submission url
+    url: "https://submit.backtrace.io/<universe>/<token>/json",
+    // Attach the files to all reports
+    attachments: attachments
```
[the remark this answers is not in the document]
Comment (perf2711) [reply packages/browser/README.md:63]: just to be sure that Ian understands:
```typescript
{
    ...
    // Submission url
    url: "https://submit.backtrace.io/<universe>/<token>/json",
    // Attach the files to all reports
    attachments,
}
```

File: packages/browser/README.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+### Breadcrumbs
+Feature in developement
+
+### Callback Methods
+The Backtrace Browser has several callback methods that allow you to modify the reports
+#### BeforeSend
+Triggers an event every time an exception in the managed environment occurs, which allows you to skip the report (by returning a null value) or to modify data that library collected before sending the report. You can use the BeforeSend event to extend attributes or JSON object data based on data the application has at the time of exception or remove the same.
+
+```js
```
Comment (perf2711) [inline packages/browser/README.md:100]: This snippet (and some others) isn't exactly written in JS - you have type annotations in `myBeforeSend`, and in `test` variable.

Can we either make two versions of each snippet for JS and TS or stick to a single language?

**Review by rick-bt — COMMENTED** [review]
Added comments

File: packages/browser/README.md (hunk 5 above)
Comment (rick-bt) [reply packages/browser/README.md:31]: Can we add these two comments to the code snippet, to direct the user to project settings?

File: packages/browser/README.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+| `agent` | string | Backtrace SDK name |
+
+#### Methods
+
+| Name | Return Type | Description |
+| - | - | - |
+| `addAttribute(attributes: Record<string, unknown>)` | void | Add attributes to the BacktraceClient reports |
+| `builder(options: BacktraceClientOptions).build()` | BacktraceClient | Sets up a new BacktraceClient for reporting|
+
+##### BacktraceClientOptions
```
Comment (rick-bt) [inline packages/browser/README.md:150]: Is this indented too far?

File: packages/browser/README.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+    const attachment3 = new BacktraceStringAttachment("CaughtErrorLog", "some error logging data here");
+    report.attachments.push(attachment3);
+    client.send(report);
+}
+```
+
+### Breadcrumbs
+Feature in developement
+
+### Application Stability Metrics
```
Comment (rick-bt) [inline packages/browser/README.md:131]: I think this makes sense to list this under features, even if it does just link to the object below. The description can also link to Sauce Docs.

File: packages/browser/README.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+The following options are available for the BacktraceClientOptions passed when initializing the BacktraceClient. 
+
+| Option Name | Type | Description | Default | Required? |
+|-|-|-|-|-|
+| `url` | String | Submission URL to send errors to | | <ul><li>- [x] </li></ul> |
+| `name` | String | Application name | | <ul><li>- [x] </li></ul> |
+| `version` | String | Application version | | <ul><li>- [x] </li></ul> |
+| `captureUnhandledErrors` | Boolean | Enable unhandled errors | `true` | <ul><li>- [ ] </li></ul> |
+| `captureUnhandledPromiseRejections` | Boolean | Enable unhandled promise rejection | `true` | <ul><li>- [ ] </li></ul> |
+| `token` | String | The submission token for error injestion.  This is required only if submitting directly to a Backtrace URL.| | <ul><li>- [ ] </li></ul> |
```
Comment (rick-bt) [inline packages/browser/README.md:161]: Should this float to 4th position?

File: packages/browser/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (ianrice07) [reply packages/browser/README.md]: Done.

File: packages/browser/README.md (hunk 1 above)
Comment (ianrice07) [reply packages/browser/README.md]: Okay.  Renaming the file name to `output_buffer`.

File: packages/browser/README.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+client.attachments.push(attachment2)
+
+// After catching an exception and generating a report
+try {
+    throw new Error("Caught exception!")
+} catch (error) {
+    const report = new BacktraceReport(error);
+    const attachment3 = new BacktraceStringAttachment("CaughtErrorLog", "some error logging data here");
+    report.attachments.push(attachment3);
+    client.send(report);
```
[the remark this answers is not in the document]
Comment (ianrice07) [reply packages/browser/README.md:131]: Updated.

File: packages/browser/README.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+    "clientID": "de6faf4d-d5b5-486c-9789-318f58a14476"
+})
+```
+
+#### Situationally in the BacktraceData or BacktraceReport
+When a BacktraceReport object is provided the attributes list can be directly modified.
+
+```ts
+const report: BacktraceReport = new BacktraceReport("My error message");
+report.attributes["myKey"] = "myValue";
```
[the remark this answers is not in the document]
Comment (ianrice07) [reply packages/browser/README.md:91]: Accepted the suggestion and updated.

File: packages/browser/README.md (hunk 8 above)
Comment (ianrice07) [reply packages/browser/README.md]: Done.

**Pull request merged**